src/devices/Common/Iot/Device/Common/WeatherHelper.cs

```diff
@@ -0,0 +1,210 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Iot.Units;
+using System;
+
+namespace Iot.Device.Common
+{
+    /// <summary>
+    /// Helpers for weather
+    /// </summary>
+    public class WeatherHelper
+    {
+        #region TemperatureAndRelativeHumidity
+        // Formulas taken from http://www.reahvac.com/tools/humidity-formulas/
+        // Extra notes for heat index taken from https://www.wpc.ncep.noaa.gov/html/heatindex_equation.shtml
+        
+        /// <summary>
+        /// The heat index (or apparent temperature) is used to measure the amount of discomfort
+        /// during the summer months when heat and humidity often combine to make it feel hotter
+        /// than it actually is. The heat index is usually used for afternoon high temperatures.
+        /// Use the summer simmer index instead for overnight low temperatures.
+        /// </summary>
+        /// <param name="airTemperature">The dry air temperature</param>
+        /// <param name="relativeHumidity">The relative humidity (RH) expressed as a percentage</param>
+        /// <returns>The heat index, also known as the apparent temperature</returns>
+        public static Temperature CalculateHeatIndex(Temperature airTemperature, double relativeHumidity)
+        {
+            double tf = airTemperature.Fahrenheit;
+            double rh = relativeHumidity;
+            double tf2 = Math.Pow(tf, 2);
+            double rh2 = Math.Pow(rh, 2);
+
+            double steadman = 0.5 * (tf + 61 + ((tf - 68) * 1.2) + (rh * 0.094));
+
+            if (steadman + tf < 160) // if the average is lower than 80F, use Steadman, otherwise use Rothfusz.
+                return Temperature.FromFahrenheit(steadman);
+
+            double rothfuszRegression = (-42.379)
+                + (2.04901523 * tf)
+                + (10.14333127 * rh)
+                - (0.22475541 * tf * rh)
+                - (6.83783 * 0.001 * tf2)
+                - (5.481717 * 0.01 * rh2)
+                + (1.22874 * 0.001 * tf2 * rh)
+                + (8.5282 * 0.0001 * tf * rh2)
+                - (1.99 * 0.000001 * tf2 * rh2);
+
+            if (rh < 13 && tf >= 80 && tf <= 112)
+            { // adjustment
```

Comment:
this comment is not exactly helpful, I'd remove it and the one below (and any similar)

Reply:
Done (pending commit)